packages/core/rendering/core-rendering-server-internal/src/rendering_service.tsx

```diff
@@ -116,6 +120,9 @@ export class RenderingService {
             catchError(() => of({}))
           )
         );
+        if ('customBranding' in renderOptions) {
+          branding = await renderOptions.customBranding.getBrandingFor(request);
+        }
```

Comment:
NIT: use the same trick we did for other setup-only service, define them as `never` in the preboot counterpart:

```ts
type RenderOptions =
  | RenderingSetupDeps
  | (RenderingPrebootDeps & { status?: never; elasticsearch?: never, customBranding?: never });
```

That way you can just normally spread as we do for other services

```ts
const { elasticsearch, http, uiPlugins, status, customBranding } = renderOptions;
```